backend/tracim_backend/applications/share/controller.py

```diff
@@ -0,0 +1,215 @@
+        """
+        remove a file share
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = ShareLib(
+            current_user=request.current_user, session=request.dbsession, config=app_config
+        )
+        api.disable_content_share(request.current_content, hapic_data.path.share_id)
+        return
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__CONTENT_FILE_ENDPOINTS])
+    @hapic.input_path(ShareTokenPathSchema())
+    @hapic.output_body(ContentShareInfoSchema())
+    def guest_download_info(
+        self, context, request: TracimRequest, hapic_data=None
+    ) -> ContentShareInContext:
+        """
+        get content file info
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = ShareLib(current_user=None, session=request.dbsession, config=app_config)
+        content_share = api.get_content_share_by_token(
+            share_token=hapic_data.path.share_token
+        )  # type: ContentShare
+
+        # TODO - G.M - 2019-08-01 - verify in access to content share can be granted
+        # we should considered do these check at decorator level
+        if content_share.content.type not in shareables_content_type:
+            raise ContentTypeNotAllowed()
+
+        return api.get_content_share_in_context(content_share)
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__CONTENT_FILE_ENDPOINTS])
+    @hapic.handle_exception(WrongSharePassword, HTTPStatus.FORBIDDEN)
+    @hapic.input_path(ShareTokenWithFilenamePathSchema())
+    @hapic.input_headers(TracimSharePasswordHeaderSchema())
+    @hapic.output_file([])
+    def guest_download_file(self, context, request: TracimRequest, hapic_data=None) -> HapicFile:
+        """
+        remove a file share
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = ShareLib(current_user=None, session=request.dbsession, config=app_config)
+        content_share = api.get_content_share_by_token(
+            share_token=hapic_data.path.share_token
+        )  # type: ContentShare
+
+        # TODO - G.M - 2019-08-01 - verify in access to content share can be granted
+        # we should considered do these check at decorator level
+        api.check_password(content_share, password=hapic_data.headers.tracim_share_password)
+        if content_share.content.type not in shareables_content_type:
+            raise ContentTypeNotAllowed()
+
+        try:
+            file = DepotManager.get().get(content_share.content.depot_file)
+        except IOError as exc:
+            raise TracimFileNotFound(
+                "file related to revision {} of content {} not found in depot.".format(
+                    content_share.content.revision_id, content_share.content.content_id
+                )
+            ) from exc
+        filename = hapic_data.path.filename
+        if not filename or filename == "raw":
```

Comment:
What is this case where filename is equal to "raw" ? (constant seem more adapted here)

Reply:
1. filename come from path. We can expect any value. In case we have raw or nothing, we do return a file with same filename as original file. In other case, we override this.
2. We do exactly the same for other get file endpoint

Reply:
I clearly don't understand the point.

Reply:
The point of the technical debt issue is: for example,as new developer, when you read this code you d'ont understand what it do. So, refactor in func with docstring, or use explicit named constant, etc.

Reply:
Explained in comment.